### PR TITLE
fix: peer validation

### LINKED ISSUE #1618: Request made to 'api/v2/node/configuration' using HTTP instead of HTTPS
**Expected Behavior**

Should use HTTPS.

**Current Behavior**

Is using HTTP, and consequentially the request is blocked by CORS.
image

**Possible Solution**

Just change the URL

**Steps to Reproduce (for bugs)**

1. Open the main screen.
2. Go to Network (cloud icon) -> Manage Networks
3. Chose the ARK Devnet
4. Click 'Save'
5. See the Error to connect to the seed server. 
6. Open the Devtools.

**Your Environment**

- Version used: Latest `develop` branch.
- Environment name and version (e.g. node.js v10.15.3): Node 12.12.0
- Operating System and version: Debian 10

### PR DESCRIPTION
**Pull request by davimello28**
## Summary

- Fixes #1618 and other possible https -> http on fetching new peers. 
- Accepts an instance of `URL()` on `validatePeer`.

## Checklist

- [X] Documentation _(if necessary)_
- [X] Tests _(if necessary)_
- [ ] Ready to be merged

### CODE REVIEW

File: src/renderer/components/Network/NetworkModal.vue [hunk 1]
```diff
@@ -468,7 +464,7 @@ export default {
 
         return success
       } catch (error) {
-        //
+        console.error(error)
```
Comment (alexbarnsley) [inline src/renderer/components/Network/NetworkModal.vue]:
```suggestion
        this.$logger.error('Could not sign message: ', error)
```

File: src/renderer/store/modules/peer.js [hunk 2]
```diff
@@ -484,27 +484,19 @@ export default {
...
-      if (!schemeUrl) {
-        baseUrl = `http://${baseUrl}`
-      }
+
       try {
-        networkConfig = await ClientService.fetchNetworkConfig(baseUrl, timeout)
+        networkConfig = await ClientService.fetchNetworkConfig(url.origin, timeout)
       } catch (error) {
-        //
+        console.error(error)
```
Comment (alexbarnsley) [inline src/renderer/store/modules/peer.js]:
```suggestion
        this._vm.$logger.error('Could not sign message: ', error)
```

File: src/renderer/store/modules/peer.js [hunk 3]
```diff
@@ -514,27 +506,27 @@ export default {
...
-      client.host = baseUrl
+      client.host = url.origin
       client.client.withOptions({ timeout: 3000 })
 
       let peerStatus
       try {
         peerStatus = await client.fetchPeerStatus()
       } catch (error) {
-        //
+        console.error(error)
```
Comment (alexbarnsley) [inline src/renderer/store/modules/peer.js]:
```suggestion
        this._vm.$logger.error('Could not sign message: ', error)
```

**Comment by alexbarnsley** [comment]
@brenopolanski could you give this a test for me please and let me know if you have any issues. Basically, test the steps in the issue which is referenced, and also try using different custom peers (e.g. https://dexplorer.ark.io). Thanks

**Comment by alexbarnsley** [comment]
@davimello28 could you fix the conflicts please

@brenopolanski don't forget to review this please

**Comment by brenopolanski** [comment]
@davimello28 @alexbarnsley 

I did some tests using the steps of the issue #1618. I realized that if you try to connect using a **"Connect custom peer"** with the following data:

Screenshot from 2020-02-13 15-40-17

It will show an error similar to this:

Screenshot from 2020-02-13 15-40-35

So close the modal, then try to connect again in ARK Devnet (on the Manage Networks page). Now it will connect and this time using `https`:

Screenshot from 2020-02-13 15-41-28

**Comment by alexbarnsley** [comment]
@brenopolanski was you on a Devnet profile when you tried the test? "Wrong nethash" should only show when connecting to a peer on a different network? E.g. a devnet peer when on a mainnet profile

**Comment by brenopolanski** [comment]
@alexbarnsley when I did the tests I was in a mainet profile.

**Comment by alexbarnsley** [comment]
@brenopolanski that error is correct then, it shouldn't be able to connect to a `https://dexplorer.ark.io` peer when on the mainnet profile. 

Was that the only thing you came across?

**Comment by brenopolanski** [comment]
@alexbarnsley 

For now, yes. I'll investigate further and see if I look for other errors.

**Comment by alexbarnsley** [comment]
@brenopolanski great, thanks. Let me know if everything works okay so I can merge this

File: src/renderer/components/Modal/ModalPeer.vue [hunk 4]
```diff
@@ -146,10 +146,12 @@ export default {
   },
 
   methods: {
-    emitConnect () {
-      this.$emit('connect', {
-        peer: this.form,
-        closeTrigger: this.closeTrigger
+    async validate () {
```
Comment (brenopolanski) [inline src/renderer/components/Modal/ModalPeer.vue]: The `validate` method is not linked to the connect button. I received the message on the browser console:

```
Property or method "emitConnect" is not defined on the instance but referenced during render...
```

Although I have updated the code, the modal does not close and does not show any alerts if the connection is successful or not.

**Comment by davimello28** [comment]
Consider #1692.

**Pull request closed without merging**